WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java

```diff
@@ -243,25 +671,120 @@ private synchronized void doNotify(long id, Notification notification) {
     }
 
     void setTotalMediaItems(PostModel post, int totalMediaItems) {
-        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
-        if (totalMediaItems <= 0) {
-            totalMediaItems = 1;
+        if (post != null) {
+            sNotificationData.totalPostItems = 1;
+            if (post.isPage()) {
+                sNotificationData.totalPageItemsIncludedInPostCount = 1;
+            }
         }
+        sNotificationData.totalMediaItems = totalMediaItems;
+    }
 
-        notificationData.totalMediaItems = totalMediaItems;
-        notificationData.itemProgressSize = 100.0f / notificationData.totalMediaItems;
+    private String buildNotificationTitleForPost(PostModel post) {
+        String postTitle = (post == null || TextUtils.isEmpty(post.getTitle())) ? mContext.getString(R.string.untitled) : post.getTitle();
+        return String.format(mContext.getString(R.string.uploading_post), postTitle);
     }
 
-    void setCurrentMediaItem(PostModel post, int currentItem) {
-        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
-        notificationData.currentMediaItem = currentItem;
+    private String buildNotificationTitleForMedia() {
+        return mContext.getString(R.string.uploading_media);
+    }
 
-        mNotificationBuilder.setContentText(String.format(mContext.getString(R.string.uploading_total),
-                currentItem, notificationData.totalMediaItems));
+    private String buildNotificationTitleForMixedContent() {
+        return mContext.getString(R.string.uploading_title);
     }
 
-    private String buildNotificationTitleForPost(PostModel post) {
-        String postTitle = TextUtils.isEmpty(post.getTitle()) ? mContext.getString(R.string.untitled) : post.getTitle();
-        return String.format(mContext.getString(R.string.uploading_post), postTitle);
+    private String buildNotificationSubtitleForPost(PostModel post){
+        String uploadingMessage = String.format(
+                mContext.getString(R.string.uploading_subtitle_posts_only),
+                sNotificationData.totalPostItems - getCurrentPostItem(),
+                (post != null && post.isPage()) ? mContext.getString(R.string.page).toLowerCase()
```

Comment:
Please don't use variables for strings like "post" or "page". This leads to problems when translating to languages where post and page have different grammatical genders. Please create two separate strings like `%1$d posts remaining` and `%1$d pages remaining` instead of `%1$d %2$s remaining`.